### ISSUE TITLE
why use the x_max in function log_sum_exp()?

### ISSUE THREAD

**Issue by huabinhuang1994** [lzx1413/PytorchSSD#22]
def log_sum_exp(x)
x_max = x.data.max()
return torch.log(torch.sum(torch.exp(x-x_max), 1, keepdim=True)) + x_max
In this function ,if we remove x_max,the output of this function is just the same,so why should we use the x_max ?

**Comment by lzx1413** [comment]
In order to handle the overflow problem. It is a common trick while coding the softmax function

**Comment by huabinhuang1994** [comment]
got it, thank you . BTW,I love your code style, it's very beatiful and easy to understand

**Comment by lzx1413** [comment]
Thanks
